src/services/document-services.ts

```diff
@@ -0,0 +1,125 @@
+import Axios, { AxiosResponse } from 'axios'
```

Comment:
I know the FeeServices imports Axios like this but I think it's bad practice.

Consider passing the existing Axios instance in as a prop, like this: https://github.com/bcgov/bcrs-shared-components/blob/e5ab4face748a4cfb6d16b1e566f6cabe88364b5/src/components/staff-comments/StaffComments.vue#L140

https://github.com/bcgov/business-filings-ui/blob/d61ffee682e1dd1b3a5931dc520e9fbb638530e4/src/components/EntityInfo/EntityMenu.vue#L6

Reply:
The only catch i bumped into here is the interceptors from a local instance may not meet all requirements. 
ie missing account-id insertion into the header etc

Work around is to create another local instance but it might also make sense to make a service or component atomic so it doesn't need to rely on that local axios etc. 

Just considerations!

Reply:
You can still provide the necessary headers in this file using a passed-in Axios instance, ie, this should still work fine:

![image](https://github.com/user-attachments/assets/192dc73c-e757-4ccf-88fc-9b17fe9f47c5)

As for interceptors, well if they're needed in the parent project then they're probably needed when this shared component is part of that project, yes?

Reply:
For sure
Just calling out that passing in axios vs a local Axios instance with defined interceptors matters. 

If a service or component here has somewhat unique headers for it's requests, than the argument could be made to just make it atomic on this end, is all.

Reply:
The thing is, I might need to use these functions in any project that has doc-related components.
However, I believe we use useBcrosFetch in Nuxt applications(e.g. dashboard ui), and in that case, I need to create a new Axios instance to pass as a prop.

Reply:
I'm debating whether the parent should manage any needed Axio instance, or whether the shared component should encapsulate that (though this feels like a "side effect").

Question: since you don't create an axios instance here, how does this work? I don't think `Axios` (TitleCase not lowercase) is an instance that you can use. Do you have this code working somewhere (eg, locally in a UI project)?

Reply:
> If a service or component here has somewhat unique headers for it's requests,

Can you think of a real-life situation like this? It's a good case for a local Axios instance...

But do we need to attach a bearer token (JWT) to DRS calls?

Reply:
First of all, I tested these functions acroos create ui, edit ui, filings ui, and sbc-auth. 
So yes I updated those repos using these functions and like I mentioned while doing this I found the error on fee-service.

Reply:
<img width="1260" alt="image" src="https://github.com/user-attachments/assets/3744a6df-0829-4eb9-acca-b2c8521ad74e" />

I implemented this approach in `Create UI` to avoid adding the JWT token to the header by default. If I need to include a specific header for its requests, I prefer passing it as a parameter to a function and adding the header within the function itself.

That said, if using a local `axios` instance is the intended approach, I’m happy to adjust by passing axios as a function parameter as well.

Reply:
Brandon, in your example above, did you use the current axios instance for the DRS service, and that's why you needed to override the interceptor?

Or were you using a separate axios instance in the DRS service? And you still needed to override the interceptor? And I still don't get how you could use axios in the service without first creating an instance.

Are we talking about separate scenarios?

Reply:
I'm still not sure whether a local Axios instance (in your component) is desirable.

I do see the problem with the unwanted interceptor. Does it actually cause a problem?

Also, is there no auth for DRS calls?

Reply:
Depending on how Axios is working currently with shared components, the local axios idea might be ideal or just as fraught with issues. I'm waiting for your response on my questions above to understand this and give an informed opinion. Thx.

Reply:
I need to override the intercepter because we get 401 response with JWT and we use apikey for authentication.

<img width="1260" alt="image" src="https://github.com/user-attachments/assets/fb98e8d0-2b02-4b99-9777-dc07e938b7e9" />

But yes I've included `axios` instance as a param for each function and I am writing unit test as requested.
Once the updated fucntions are fully test locally across create ui, edit ui, filings ui, and sbc-auth, i will make another commit and will inform you.

Thanks.

Reply:
Are you saying that `import Axios, { AxiosResponse } from 'axios'` returns the existing instance (with interceptors), and that's why you don't need to "create" it but you do need to override the interceptor?

In a way, that's good, as it means we don't have to pass the axios instance into the component; the code above just returns it.

In a way, that's bad, because the existing instance can have interceptors, and it's really unfortunate you need to hack the UI's axios instance in order to use an external component (which, ideally, would be "atomic" as Cameron put it, or I would say "independent").

If the above is true then you don't need to pass in the prop, and let's keep this simple for now -- I will accept your existing code.

If I still have my lines crossed, let me know 😄

Reply:
Happy Friday Severin!

I just pushed a new commit which includes the below updates:
- passed axios instance as a param
- implemented unit test
- confirmed  fee-summary component is working on storybook
- tested on create-ui and edit-ui
- addressed additional issues.

This is the first time for me to work on npm package which is a great opportunity to learn new skills.
Please let me know if I missed anything.
Thank you for your help! :heart: :heart: :heart: